### PR TITLE
[clang] NFCI: Use `FileEntryRef` for `FileID` creation

### PR DESCRIPTION
**Pull request by jansvoboda11**
This patch removes the `SourceManager` APIs that create `FileID` from a `const FileEntry *` in favor of APIs that take `FileEntryRef`. This also removes a misleading documentation that claims `nullptr` file entry represents stdin. I don't think that's right, since we just try to dereference that pointer anyways.

### CODE REVIEW

File: clang/lib/Basic/SourceManager.cpp [hunk 1]
```diff
@@ -527,17 +527,6 @@ FileID SourceManager::getNextFileID(FileID FID) const {
 
 /// Create a new FileID that represents the specified file
 /// being \#included from the specified IncludePosition.
-///
-/// This translates NULL into standard input.
-FileID SourceManager::createFileID(const FileEntry *SourceFile,
```
Comment (benlangmuir) [inline clang/lib/Basic/SourceManager.cpp:532]: Were there no callers that could pass `NULL`?

File: clang/tools/clang-rename/ClangRename.cpp [hunk 2]
```diff
@@ -228,7 +228,7 @@ int main(int argc, const char **argv) {
 
     Tool.applyAllReplacements(Rewrite);
     for (const auto &File : Files) {
-      auto Entry = FileMgr.getFile(File);
+      auto Entry = FileMgr.getOptionalFileRef(File);
```
Comment (benlangmuir) [inline clang/tools/clang-rename/ClangRename.cpp:231]: Another place we could use `getFileRef` and print the error message.

File: clang/lib/Basic/SourceManager.cpp (hunk 1 above)
Comment (jansvoboda11) [reply clang/lib/Basic/SourceManager.cpp:532]: Not that I could find.

**Comment by jansvoboda11** [comment]
Thanks for the feedback. I used `llvm::cantFail(FM.getFileRef(Name))` in some places to fail more gracefully. The error-reporting changes you suggest I'd like to put into a follow-up patch, to keep this one as NFC as possible. WDYT?

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**